From 1481a372f5323234329546235d835037bf8390bd Mon Sep 17 00:00:00 2001
From: The-BB <tun.chen.bo@gmail.com>
Date: Sun, 11 Dec 2022 12:13:46 +0300
Subject: [PATCH] include/package-ipkg.mk: autofix exec path in scripts

---
 include/package-ipkg.mk | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/include/package-ipkg.mk b/include/package-ipkg.mk
index 3acd50c7..962d8182 100644
--- a/include/package-ipkg.mk
+++ b/include/package-ipkg.mk
@@ -208,6 +208,13 @@ $(_endef)
 	$(if $(Package/$(1)/install-overlay),mkdir -p $(PACKAGE_DIR) $$(IDIR_$(1))/rootfs-overlay)
 	$(call Package/$(1)/install-overlay,$$(IDIR_$(1))/rootfs-overlay)
 	-find $$(IDIR_$(1)) -name 'CVS' -o -name '.svn' -o -name '.#*' -o -name '*~'| $(XARGS) rm -rf
+	-find $$(IDIR_$(1)) -type f -exec $(SED) \
+		's,^#!.*bash,#!/opt/bin/bash,' -e \
+		's,^#!.*node$$$$,#!/opt/bin/node,' -e \
+		's,^#!.*perl,#!/opt/bin/perl,' -e \
+		's,^#!.*python,#!/opt/bin/python,' -e \
+		's,^#!.*ruby,#!/opt/bin/ruby,' -e \
+		's,^#!.*env sh,#!/bin/sh,' {} +
 	@( \
 		find $$(IDIR_$(1)) -name lib\*.so\* -or -name \*.ko | awk -F/ '{ print $$$$NF }'; \
 		for file in $$(patsubst %,$(PKG_INFO_DIR)/%.provides,$$(IDEPEND_$(1))); do \
-- 
2.30.2

